dist/js/components/AmountWidget.js

```diff
@@ -1,70 +1,63 @@
 import { select, settings } from "../settings.js";
+import BaseWidget from "./BaseWidget.js";
 
-class AmountWidget {
+class AmountWidget extends BaseWidget {
   constructor(element) {
+    super(element, settings.amountWidget.defaultValue);
+
     const thisWidget = this;
+
     // console.log('AmountWidget:', thisWidget);
     // console.log('constructor arguments:', element);
+
     thisWidget.getElements(element);
-    thisWidget.setValue(settings.amountWidget.defaultValue);
+    // thisWidget.setValue(thisWidget.input.value || settings.amountWidget.defaultValue);
```

Comment:
Nie możesz usunąć ustawiania początkowej wartości, bo `AmountWidget` nie będzie działał prawidłowo. Możesz sprawdzić, że teraz w menu nie ma wartości początkowej, a po kliknięciu `+` pokazuje się od razu `2`.
Sposób, w jaki początkowa wartość jest tutaj ustawiana, jest dobry. Problemem jest to, że `thisWidget.input` jest `undefined`.
Jeśli wylogujesz sobie w consoli `thisWidget`, to będziesz mogła sprawdzić, gdzie jest właściwość `input` i jak się do niej odwołać.